### DISCUSSION TITLE
When rendering of split headers when using a data in the Split By field, the re-rendering on table.update adds rows

### DISCUSSION THREAD

**Discussion by ramirezg-web** [perspective-dev/perspective#2519] [Q&A]
## Bug Report
When rendering of split headers when using a data in the Split By field, Perspective table.update() adds rows instead of re-rendering table with updated values

### Steps to Reproduce:
 1. Find replication of issue here: Reproduction:https://codesandbox.io/p/sandbox/serene-albattani-h96wxj
Use Perspective Viewer to render the table
 2. await 3 seconds per the setTimeout function
 3. the table adds rows

### Expected Result:
 Expected to see table render with updated values

### Actual Result:
Expected to see updated values, but new rows were added.
Reproduction:https://codesandbox.io/p/sandbox/serene-albattani-h96wxj

### Environment:
Perspective 2.7.1
MacOS Ventura 13.6.1
Chrome Browser 120.0.6099.234 (Official Build) (x86_64)

**Comment by timkpaine** [comment]
Your initial data is created on the 22nd
```javascript
        const table = await worker.table(createData(22), {
```

But you update with data from the 26

```javascript
          const newData = createData(26);
```

So I think the behavior is correct? Please let me know if I'm reading this wrong

**Comment by ramirezg-web** [comment]
Hello, thanks for your reply. We would like to seek clarity on Table.update method. What is the expected behavior? What is the expected behavior when we have an index parameter, as we have a parameter in this case: "timestamp". Does it replace the data or append rows to the data. If we are adding different dates, our impression is that the values of timestamp should be different, but all it does is add rows. Should we use table.replace instead? I've included screenshots. Please note that the secondary screenshot is for enhanced viewership with an expanded column. 

![table update original data](https://github.com/finos/perspective/assets/68044078/48584fc9-0a7b-4ba2-bc35-1072e09acfcf)
![expanded for viewership](https://github.com/finos/perspective/assets/68044078/87e56fbe-2e50-4624-9a7e-9c311ced5d81)
![on updateData, new rows are added instead of data replaced](https://github.com/finos/perspective/assets/68044078/ade80b36-532c-4eb5-afa8-298133e2c57a)
Repo Replication 

https://codesandbox.io/p/sandbox/friendly-sunset-h96wxj
